Obsidian.API/Noise/VoronoiBiomes.cs

```diff
@@ -79,22 +81,50 @@ public override double GetValue(double x, double y, double z)
                     BaseBiome = 0.0,
                     Biome = 0
                 };
-
-                cells[index++] = cell;
             }
         }
 
-        MemoryExtensions.Sort(cells, (a, b) => { return a.DistanceToPoint > b.DistanceToPoint ? 1 : -1; });
-        var meVal = NoiseGenerator.ValueNoise3D((int)Math.Floor(cells[0].Point.x), 0, (int)Math.Floor(cells[0].Point.z));
-        (cells[0].BaseBiome, cells[0].Variant) = GetBaseBiome(meVal);
-        var nearestVal = NoiseGenerator.ValueNoise3D((int)Math.Floor(cells[1].Point.x), 0, (int)Math.Floor(cells[1].Point.z));
-        (cells[1].BaseBiome, cells[1].Variant) = GetBaseBiome(nearestVal);
+        VoronoiCell me, nearest;
+        Unsafe.SkipInit(out me);
+        Unsafe.SkipInit(out nearest);
+        GetMin(cells, ref me, ref nearest);
 
+        var meVal = NoiseGenerator.ValueNoise3D((int)Math.Floor(me.Point.x), 0, (int)Math.Floor(me.Point.z));
+        (me.BaseBiome, me.Variant) = GetBaseBiome(meVal);
+        var nearestVal = NoiseGenerator.ValueNoise3D((int)Math.Floor(nearest.Point.x), 0, (int)Math.Floor(nearest.Point.z));
+        (nearest.BaseBiome, nearest.Variant) = GetBaseBiome(nearestVal);
+
+        return (double)ProcessBiomeRules(me, nearest);
+    }
+
+    private static void GetMin(ReadOnlySpan<VoronoiCell> cells, ref VoronoiCell min, ref VoronoiCell secondMin)
```

Comment:
Curious. I don't understand why this is more optimized than `Sort()`

Reply:
`O(n log n)` → `O(n)`